index.js

```diff
@@ -281,3 +281,11 @@ export function hasProperty(object, path) {
 
 	return true;
 }
+
+export function escapePath(path) {
+	if (typeof path !== 'string') {
+		throw new TypeError('Expected a string');
+	}
+
+	return path.replace(/[\\.[]/g, '\\$&');
```

Comment:
Are you sure there's no way to trick this regex? I want to be confident that this doesn't introduce any security vulnerability.

Reply:
The regex is so short that it seems infeasible for any exploitation. Badly declared capture groups can sometimes cause catastrophic backtracking but this regex has not a single one.

Reply:
I was more thinking of ways to circumvent the character escaping.

Reply:
Not that I can think of, but I could replace the regex with a regular scan-and-replace system in case that is necessary.

Reply:
Let's just go with what's here for now.